akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventEnvelopeSpec.scala

```diff
@@ -77,6 +81,9 @@ class EventEnvelopeSpec extends AnyWordSpecLike with Matchers {
       env4.metadata[java.time.Instant] shouldBe Some(instant2)
       env4.eventMetadata shouldBe Some(instant2) // deprecated
 
+      env.removeMetadata[String].metadata[String] shouldBe None
+      env.removeMetadata[AnyRef] shouldBe theSameInstanceAs(env)
```

Comment:
Here we need to remove from env4 and it will fail because using `filter` instead of `filterNot`.

Reply:
Ah, that's why, thx